ts/webui/src/components/managementExp/ExperimentManager.tsx

```diff
@@ -312,50 +312,45 @@ class Experiment extends React.Component<{}, ExpListState> {
      */
     private commonSelectString = (data: AllExperimentList[], field: string): AllExperimentList[] => {
         const { selectedStatus, selectedPlatform, selectedStartDate, selectedEndDate } = this.state;
-        const hasStatus = selectedStatus === '' ? false : true;
-        const hasPlatform = selectedPlatform === '' ? false : true;
-        const hasStartDate = selectedStartDate === undefined ? false : true;
-        const hasEndDate = selectedEndDate === undefined ? false : true;
 
         if (field === 'status') {
-            if (hasPlatform) {
-                data = filterByStatusOrPlatform(selectedPlatform, 'platform', data);
-            }
+            data = filterByStatusOrPlatform(selectedPlatform, 'platform', data);
         }
         if (field === 'platform') {
-            if (hasStatus) {
-                data = filterByStatusOrPlatform(selectedStatus, 'status', data);
-            }
+            data = filterByStatusOrPlatform(selectedStatus, 'status', data);
         }
 
         if (field === '') {
-            if (hasPlatform) {
-                data = filterByStatusOrPlatform(selectedPlatform, 'platform', data);
-            }
-            if (hasStatus) {
-                data = filterByStatusOrPlatform(selectedStatus, 'status', data);
-            }
+            data = Array.from(
+                new Set([
+                    ...filterByStatusOrPlatform(selectedPlatform, 'platform', data),
+                    ...filterByStatusOrPlatform(selectedStatus, 'status', data)
+                ])
+            );
         }
 
-        if (hasStartDate) {
-            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            data = data.filter(temp => compareDate(new Date(temp.startTime), selectedStartDate!));
-        }
-        if (hasEndDate) {
-            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            data = data.filter(temp => compareDate(new Date(temp.endTime), selectedEndDate!));
-        }
+        data = data.filter(
+            item =>
+                (selectedStartDate !== undefined ? compareDate(new Date(item.startTime), selectedStartDate) : true) &&
```

Comment:
`(selectedStartDate !== undefined || compareDate(...))` looks more straightforward to me

Reply:
Indeed.